Temp

RimvydasVainutisCS opened the pull request.

rvainutis committed:
202201281510

rvainutis committed:
202201301625

rvainutis committed:
202201302046

RimvydasVainutisCS committed:
override ToString()

RimvydasVainutisCS committed:
override Equals()

RimvydasVainutisCS committed:
override Equals() full

RimvydasVainutisCS committed:
IVersionable interface implementation.

RimvydasVainutisCS committed:
ICloneable implementacija

RimvydasVainutisCS committed:
Icloneable interface implementation.

RimvydasVainutisCS committed:
IVersionable GetVersion nebaigtas

RimvydasVainutisCS committed:
small update

RimvydasVainutisCS committed:
Lesson 1 clone with a training type change: Text => Video

RimvydasVainutisCS committed:
kodas kodelis

RimvydasVainutisCS committed:
Merge branch 'master' into temp

IgorKheidorov commented on Net011/LinkToNetRsc.cs:
```diff
@@ -1,20 +1,36 @@
+            if (!(obj is LinkToNetRsc))
+            {
+                return false;
+            }
```
Use (obj as LinkToNetRsc) cast, not to throw the exception

IgorKheidorov commented on Net011/TextMaterial.cs:
```diff
@@ -27,11 +22,29 @@ public string Text
+        public override string ToString()
+        {
+            return Description;
+        }
```
Simplify, one if

IgorKheidorov commented on Net011/TrainingLesson.cs:
```diff
@@ -2,23 +2,83 @@
+            }
+            return (Id == ((TrainingMaterial)obj).Id);
+        }
+
```
Why 0? )))

RimvydasVainutisCS replied:
This line is added automatically by the compiler so that I won't get a "warning" message in the Error list :D I have no idea what is the real solution for the GetHashCode() method here.

IgorKheidorov commented on Net011/VideoMaterial.cs:
```diff
@@ -40,11 +42,40 @@ public string SplashScreenUri
+        {
+            return _version;
+        }
+
```
Check for size

IgorKheidorov commented on Net011/TrainingLesson.cs:
```diff
@@ -2,23 +2,83 @@
+            var clone = new TrainingLesson
+            {
+                Id = Id,
+                Description = Description,
```
Do not copy links, deep cloning has to be

IgorKheidorov commented on Net011/TrainingEntity.cs:
```diff
@@ -5,10 +5,10 @@ namespace Net011
+            set;
         }
         public string Description
         {
```
1. Change the order of checks, if value is null you will have exception in line 21
2. Use const int value instead of 256, magic number

RimvydasVainutisCS committed:
Byte array size check added.
obj as LinkToNetRsc changed.

RimvydasVainutisCS committed:
hardcoded value of 256 changed to a constant
no more dirty magic numbers!

RimvydasVainutisCS committed:
if statement simplified, using only one

RimvydasVainutisCS committed:
small update
found couple hardcoded magic numbers, changed to constants,
updated the messages to use these constants.

RimvydasVainutisCS committed:
deep copy added (hopefully) :)

RimvydasVainutisCS committed:
changed is obj -> as obj
updated GetHashCode()
